chat-widget/automation_tests/e2e/pages/base.page.ts

```diff
@@ -0,0 +1,97 @@
+import { Page } from "playwright";
+import fs from "fs";
+import { TestSettings } from "../../configuration/test-settings";
+import { CustomLiveChatWidgetConstants } from "e2e/utility/constants";
+
+export class BasePage {
+    protected isBaseClosed = false;
+    constructor(protected _page: Page) {}
+
+    public get Page(): Page {
+        return this._page;
+    }
+
+    public set Page(page: Page) {
+        this._page = page;
+    }
+
+    public async closePage() {
+        if (this.isBaseClosed) {
+            return;
+        }
+        await this._page.close();
+        this.isBaseClosed = true;
+    }
+
+    public async scrollTo(selector: string) {
+        await this.Page.evaluate(
+            (param) => {
+                document.querySelector(param.selector).scrollIntoView();
+            },
+            { selector: selector }
+        );
+    }
+
+    public async openLiveChatWidget() {
+        const path = fs.realpathSync(CustomLiveChatWidgetConstants.CustomLiveChatWidgetFilePath);
+        await this.Page.goto("file://" + path, { waitUntil: "domcontentloaded" });
+    }
+
+    public async openLiveChatAdaptiveCardWidget() {
+        const path = fs.realpathSync(CustomLiveChatWidgetConstants.CustomLCWAdaptiveCardWidgetFilePath);
+        await this.Page.goto("file://" + path, { waitUntil: "domcontentloaded" });
+    }
+
+    public async openLiveChatAdaptiveCardWrapWidget() {
+        const path = fs.realpathSync(CustomLiveChatWidgetConstants.CustomLCWAdaptiveCardWrapWidgetFilePath);
+        await this.Page.goto("file://" + path, { waitUntil: "domcontentloaded" });
+    }
+
+    public async openLiveChatAttachmentwithPrechatWidget() {
+        const path = fs.realpathSync(CustomLiveChatWidgetConstants.CustomLCWAttachmentPrechatWidgetFilePath);
+        await this.Page.goto("file://" + path, { waitUntil: "domcontentloaded" });
+    }
+
+    public async openLiveChatBotWidget() {
+        const path = fs.realpathSync(CustomLiveChatWidgetConstants.CustomLCWBotWidgetFilePath);
+        await this.Page.goto("file://" + path, { waitUntil: "domcontentloaded" });
```

Comment:
create one common method with filePath as parameter.
e..g 

public async openLiveChatWidget(filePath: string) {
const path = fs.realpathSync(filePath);
        await this.Page.goto("file://" + path, { waitUntil: "domcontentloaded" });
}

and call like below,
// open live chat bot widget 
openLiveChatWidget(CustomLiveChatWidgetConstants.CustomLCWBotWidgetFilePath);

// open live chat with attachment and prechat widget
openLiveChatWidget(CustomLiveChatWidgetConstants.CustomLCWAttachmentPrechatWidgetFilePath)

Reply:
Created the common method.